btf/traversal.go

```diff
@@ -40,8 +40,9 @@ func children(typ Type, yield func(child *Type) bool) bool {
 	// Explicitly type switch on the most common types to allow the inliner to
 	// do its work. This avoids allocating intermediate slices from walk() on
 	// the heap.
+	var tags []string
 	switch v := typ.(type) {
-	case *Void, *Int, *Enum, *Fwd, *Float:
+	case *Void, *Int, *Enum, *Fwd, *Float, *declTag:
```

Comment:
If we have to deal with pre-existing `declTag`s, we should probably recurse into `.Type`.

*Pre-existing*: a user obtains a reference to a `declTag` somehow and submits it for marshalling.

Reply:
The thought is that we essentially make *declTag an internal type, it shouldn't exist since all tags are part of other types. We create them during traversal as if they hang under the other types (inverted from how they are encoded).

This way, we only traverse decl tags of types that are actually in use then we go an marshal BTF blob

Reply:
Yes. But since a user can't create a  `declTag` from scratch, they would have to first marshal a BTF blob, then guess the correct ID, get the `btf.Type` and add it to the builder. That would result in a `no ID for type` panic. But that would likely happen with any type you do this with that references another.

Reply:
We could consider returning an error if the requested ID is for a declTag, but I think that would cause unnecessary confusion and complexity. As long as the happy path is reasonably free of footguns, it's fine, and I think that's the case here.